pydantic_ai_slim/pydantic_ai/models/mistral.py

```diff
@@ -608,7 +608,7 @@ def _try_get_result_tool_from_text(text: str, result_tools: dict[str, ToolDefini
                     continue
 
                 # The following part_id will be thrown away
-                return ToolCallPart(tool_name=result_tool.name, args=output_json)
+                return ToolCallPart(tool_name=result_tool.name, args=output_json, tool_call_id=_generate_tool_call_id())
```

Comment:
```suggestion
                return ToolCallPart(tool_name=result_tool.name, args=output_json)
```
I think it's set by the factory.